src/fi_tostr.c

```diff
@@ -77,12 +79,15 @@ static void fi_remove_comma(char *buffer)
 
 static void strcatf(char *dest, const char *fmt, ...)
 {
-	size_t len = strlen(dest);
+        size_t len = strnlen(dest,FI_BUFSIZ);
 	va_list arglist;
 
 	va_start (arglist, fmt);
-	vsnprintf(&dest[len], BUFSIZ - 1 - len, fmt, arglist);
+	vsnprintf(&dest[len], FI_BUFSIZ - 1 - len, fmt, arglist);
 	va_end (arglist);
+
+	if (strnlen(dest, FI_BUFSIZ) >= FI_BUFSIZ-2)
+	  sprintf(&dest[FI_BUFSIZ-3], "…\n");
```

Comment:
I'd even checked this although clearly not very well.  It seems it can be one byte but is longer in UTF-8, hexdump of the source is showing three bytes used so yes, this code is wrong.